app/src/main/java/com/github/lookupgroup27/lookup/ui/profile/Profile.kt

```diff
@@ -63,44 +66,64 @@ fun ProfileScreen(navigationActions: NavigationActions, avatarViewModel: AvatarV
                   Modifier.fillMaxSize()
                       .padding(horizontal = if (isLandscape) 16.dp else 0.dp)
                       .padding(top = if (isLandscape) 24.dp else 172.dp)
-                      .verticalScroll(rememberScrollState()),
+                      .verticalScroll(rememberScrollState()), // Make the column scrollable
               horizontalAlignment = Alignment.CenterHorizontally,
               verticalArrangement = Arrangement.Top) {
-                // Profile Icon
-                val avatarRes = selectedAvatar ?: R.drawable.default_profile_icon
-                Icon(
-                    painter = painterResource(id = avatarRes),
-                    contentDescription = "Profile Icon",
-                    modifier = Modifier.size(150.dp),
-                    tint =
-                        if (avatarRes == R.drawable.default_profile_icon) Color.White
-                        else Color.Unspecified)
+                // Profile Icon Box for FAB placement
+                Box(contentAlignment = Alignment.BottomEnd) {
+                  // Profile Icon
+                  val avatarRes = selectedAvatar ?: R.drawable.default_profile_icon
+                  Icon(
+                      painter = painterResource(id = avatarRes),
+                      contentDescription = stringResource(R.string.profile_icon_description),
+                      modifier = Modifier.size(150.dp).padding(bottom = 8.dp),
+                      tint =
+                          if (avatarRes == R.drawable.default_profile_icon) Color.White
+                          else Color.Unspecified)
 
-                Button(
-                    onClick = { navigationActions.navigateTo(Screen.AVATAR_SELECTION) },
-                    colors = ButtonDefaults.buttonColors(DarkPurple)
-                    // modifier = Modifier.align(Alignment.Center)
-                    ) {
-                      Text("Change Avatar")
-                    }
+                  // Show FAB when no avatar is selected
+                  if (selectedAvatar == null || selectedAvatar == R.drawable.default_profile_icon) {
+                    FloatingActionButton(
+                        onClick = { navigationActions.navigateTo(Screen.AVATAR_SELECTION) },
+                        containerColor = DarkPurple,
+                        modifier =
+                            Modifier.size(36.dp) // Smaller size for Instagram-like style
+                                .offset(x = (-8).dp, y = (-8).dp) // Adjust position slightly
+                        ) {
+                          Icon(
+                              imageVector = Icons.Filled.Add,
+                              contentDescription = stringResource(R.string.add_avatar),
+                              tint = Color.White)
+                        }
```

Comment:
You could consider wrapping this code in a small composable function to make it clear that it represents the Floating Action Button (FAB) for adding an avatar. This would improve readability and make the purpose of the code more evident.

Reply:
Great idea, thank you for this suggestion.